commit 6a4d21a7664b3b2a62b7bf9fbd0b89b52c30acef
Author: Leon Romanovsky <leonro@mellanox.com>
Date:   Sun Jul 1 19:36:24 2018 +0300

    RDMA/i40w: Hold read semaphore while looking after VMA
    
    [ Upstream commit 5d9a2b0e28759e319a623da33940dbb3ce952b7d ]
    
    VMA lookup is supposed to be performed while mmap_sem is held.
    
    Fixes: f26c7c83395b ("i40iw: Add 2MB page support")
    Signed-off-by: Leon Romanovsky <leonro@mellanox.com>
    Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/infiniband/hw/i40iw/i40iw_verbs.c b/drivers/infiniband/hw/i40iw/i40iw_verbs.c
index 68679ad4c6da..937899fea01d 100644
--- a/drivers/infiniband/hw/i40iw/i40iw_verbs.c
+++ b/drivers/infiniband/hw/i40iw/i40iw_verbs.c
@@ -1409,6 +1409,7 @@ static void i40iw_set_hugetlb_values(u64 addr, struct i40iw_mr *iwmr)
 	struct vm_area_struct *vma;
 	struct hstate *h;
 
+	down_read(&current->mm->mmap_sem);
 	vma = find_vma(current->mm, addr);
 	if (vma && is_vm_hugetlb_page(vma)) {
 		h = hstate_vma(vma);
@@ -1417,6 +1418,7 @@ static void i40iw_set_hugetlb_values(u64 addr, struct i40iw_mr *iwmr)
 			iwmr->page_msk = huge_page_mask(h);
 		}
 	}
+	up_read(&current->mm->mmap_sem);
 }
 
 /**
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1118_6a4d21a7664b3b2a62b7bf9fbd0b89b52c30acef.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1118_6a4d21a7664b3b2a62b7bf9fbd0b89b52c30acef.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
